ambry-api/src/main/java/com/github/ambry/store/Store.java

```diff
@@ -102,6 +103,16 @@
    */
   MessageInfo findKey(StoreKey key) throws StoreException;
 
+  /**
+   * Return a {@link Map} of {@code storeKeys} to {@link MessageInfo} of the keys.
+   * This method will only be used in replication thread, and the map returned will contain message infos for only those
+   * keys that are found in the local store.
+   * @param storeKeys The list of keys for which to return {@link MessageInfo}s.
+   * @return The {@link Map} of key to {@link MessageInfo}.
+   * @throws StoreException
+   */
+  Map<StoreKey, MessageInfo> findKeys(List<? extends StoreKey> storeKeys) throws StoreException;
```

Comment:
If this method and approach is acceptable, we no longer need `findKey()` to be a public method, and hence it can be removed from the interface.

Reply:
Yes, I would remove it.

Reply:
done.